dnfdaemon-client/main.cpp

```diff
@@ -170,7 +170,19 @@ static bool parse_args(Context & ctx, int argc, char * argv[]) {
 }  // namespace dnfdaemon::client
 
 int main(int argc, char * argv[]) {
-    auto connection = sdbus::createSystemBusConnection();
+    std::unique_ptr<sdbus::IConnection> connection;
+
+    try{
+        connection = sdbus::createSystemBusConnection();
+    } catch (const sdbus::Error & ex) {
+        std::cerr << ex.getMessage() << std::endl;
+        std::cerr << "Is D-Bus daemon running?" << std::endl;
+        return 1;
+    } catch (...) {
```

Comment:
Please don't do this. You're only catching `sdbus::Error` above this, so the `...` here can catch any valid exception and you'll just log that non-descriptive message below.

For the case the code would throw something not even inheriting from standard exceptions I'd prefer not to catch that at all, because again, you're just hiding all details and we won't know what happened at all. If you don't catch the exception, a `terminate()` will be called and the core will be dumped, meaning we'll get all the information we can about the problem.